lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -245,8 +245,8 @@ defmodule Kernel.ParallelCompiler do
     case state.each_cycle.() do
       [] ->
         modules = write_module_binaries(output, beam_timestamp, result)
-        warnings = Enum.reverse(warnings)
-        {:ok, modules, warnings}
+        warnings = Enum.reverse(warnings) ++ check_modules(modules)
```

Comment:
In the future, `check_modules` will return updated bytecodes and warnings. I am not sure if you want to change this now, but I thought I would comment on it.

Reply:
Also, prolly doesn't matter but

```suggestion
        warnings = Enum.reverse(warnings, check_modules(modules))
```